js/service/shell/captureio.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2014-2015 runtime.js project authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var events = require('events');
+exports.events = new events.EventEmitter();
```

Comment:
CaptureIO is basically that, except in a different file so it's easier to manage. `runCommand` creates a new CaptureIO for each command because they each have properties belonging to them (ex: `readln`). PLUS, it has events that are returned to whatever called `runCommand`. That's why I made it.